server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -69,9 +71,25 @@
      */
     protected FieldMapper(String simpleName, MappedFieldType mappedFieldType,
                           MultiFields multiFields, CopyTo copyTo) {
-        this(simpleName, mappedFieldType, Collections.emptyMap(), multiFields, copyTo);
+        this(simpleName, mappedFieldType, Collections.emptyMap(), multiFields, copyTo, false, null);
     }
 
+    /**
+     * Create a FieldMapper with no index analyzers
+     * @param simpleName        the leaf name of the mapper
+     * @param mappedFieldType   the MappedFieldType associated with this mapper
+     * @param multiFields       sub fields of this mapper
+     * @param copyTo            copyTo fields of this mapper
+     * @param hasScript         whether a script is defined for the field
+     * @param onScriptError     the behaviour for when the defined script fails at runtime
+     */
+    protected FieldMapper(String simpleName, MappedFieldType mappedFieldType,
+                          MultiFields multiFields, CopyTo copyTo,
+                          boolean hasScript, String onScriptError) {
+        this(simpleName, mappedFieldType, Collections.emptyMap(), multiFields, copyTo, hasScript, onScriptError);
+    }
```

Comment:
if we agree on the new constructors, I think we should try to move all the callers of the ones without the additional parameters to these, otherwise we end up with too many constructors

Reply:
I like pulling out the common behaviour, but maybe we should try and localise this more? Have an intermediate class called `ScriptableFieldMapper` or something like that?

Reply:
I am not sure it's worth the complexity of one additional intermediate base class, especially as more types will support a script.

Reply:
Fair enough - we can probably merge the single analyzer and multiple analyzer constructors at least without causing too much noise.

Reply:
agreed, let's do it as a followup?